set.go

```diff
@@ -158,17 +166,43 @@ func (s *Set) makeElemList(vals []SetElement) ([]netlink.Attribute, error) {
 			item = append(item, netlink.Attribute{Type: unix.NFTA_SET_ELEM_FLAGS | unix.NLA_F_NESTED, Data: binaryutil.BigEndian.PutUint32(flags)})
 		}
 
-		encodedKey, err := netlink.MarshalAttributes([]netlink.Attribute{{Type: unix.NFTA_SET_ELEM_KEY, Data: v.Key}})
+		encodedKey, err := netlink.MarshalAttributes([]netlink.Attribute{{Type: unix.NFTA_DATA_VALUE, Data: v.Key}})
 		if err != nil {
 			return nil, fmt.Errorf("marshal key %d: %v", i, err)
 		}
 		item = append(item, netlink.Attribute{Type: unix.NFTA_SET_ELEM_KEY | unix.NLA_F_NESTED, Data: encodedKey})
-
-		if len(v.Val) > 0 {
-			encodedVal, err := netlink.MarshalAttributes([]netlink.Attribute{{Type: unix.NFTA_SET_ELEM_DATA, Data: v.Val}})
+		// There are 3 possible cases for Data, Set element Data (no data), Map Element Data and Vmap Element Data
```

Comment:
This comment will be easier to read if you split it across multiple lines:

```go
// There are 3 possible cases for Data:
//
// 1. Set Element Data (no data)
// 2. Map Element Data
// 3. Vmap Element Data
```

Also, the `switch` statement below has only 2 cases, not 3. I assume the first `case` statement refers to 3, and the second `case` statement to 2? Can you add comments to make this clear please?